crmd/cib.c

```diff
@@ -68,20 +68,39 @@ do_cib_updated(const char *event, xmlNode * msg)
 
     crm_element_value_int(patchset, "format", &format);
     if (format == 1) {
-        if (get_xpath_object
-            ("//" F_CIB_UPDATE_RESULT "//" XML_TAG_DIFF_ADDED "//" XML_CIB_TAG_CRMCONFIG, msg,
-             LOG_TRACE) != NULL) {
+        if ((get_xpath_object(
+                 "//" F_CIB_UPDATE_RESULT "//" XML_TAG_DIFF_ADDED "//" XML_CIB_TAG_CRMCONFIG,
+                 msg, LOG_TRACE) != NULL) ||
+            (get_xpath_object(
+                 "//" F_CIB_UPDATE_RESULT "//" XML_TAG_DIFF_ADDED "//" XML_CIB_TAG_NOTIFICATIONS,
+                 msg, LOG_TRACE) != NULL)) {
             mainloop_set_trigger(config_read);
         }
-
     } else if (format == 2) {
         for (change = __xml_first_child(patchset); change != NULL; change = __xml_next(change)) {
             const char *xpath = crm_element_value(change, XML_DIFF_PATH);
-            if (xpath != NULL
-                && strstr(xpath, "/" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION "/" XML_CIB_TAG_CRMCONFIG "/")) {
-                mainloop_set_trigger(config_read);
-                break;
+
+            if (xpath == NULL) {
+                continue;
             }
+		
+            /* modifying properties */
+            if (!strstr(xpath, "/" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION "/" XML_CIB_TAG_CRMCONFIG "/") &&
+                !strstr(xpath, "/" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION "/" XML_CIB_TAG_NOTIFICATIONS)) {
```

Comment:
FYI, one thing I've started doing recently is defining constants for xpath fragments, e.g.

#define XPATH_CONFIGURATION "/" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION

It definitely makes the code look nicer, but I don't know yet whether it will be easier or harder to follow when I come back to look at it in 6 months.